pkg/config/engine/containerd/config_v2.go

```diff
@@ -60,6 +68,14 @@ func (c *Config) AddRuntime(name string, path string, setAsDefault bool, configO
 		config.SetPath([]string{"plugins", "io.containerd.grpc.v1.cri", "containerd", "default_runtime_name"}, name)
 	}
 
+	// If the systemd cgroup driver is required, and no other setting is applied, we set SystemdCgroup.
+	if c.CgroupDriver == cgroupDriverSystemd {
+		valueSet := config.GetPath([]string{"plugins", "io.containerd.grpc.v1.cri", "containerd", "runtimes", name, "options", "SystemdCgroup"})
+		if valueSet == nil {
+			config.SetPath([]string{"plugins", "io.containerd.grpc.v1.cri", "containerd", "runtimes", name, "options", "SystemdCgroup"}, true)
+		}
+	}
```

Comment:
Same comment as above

Reply:
I have moved the systemd-specifics to #578 to keep this PR more focussed.